diff --git ui/views/controls/menu/menu_host.cc ui/views/controls/menu/menu_host.cc
index 865b7dac0b8f..7ed892dfbc94 100644
--- ui/views/controls/menu/menu_host.cc
+++ ui/views/controls/menu/menu_host.cc
@@ -130,7 +130,11 @@ void MenuHost::InitMenuHost(Widget* parent,
   // Activatable, so that calling Show in ShowMenuHost will
   // get keyboard focus.
   if (parent == nullptr)
+#if defined(OS_LINUX)
+    params.activatable = Widget::InitParams::ACTIVATABLE_NO;
+#else
     params.activatable = Widget::InitParams::ACTIVATABLE_YES;
+#endif
 #if defined(OS_WIN)
   // On Windows use the software compositor to ensure that we don't block
   // the UI thread blocking issue during command buffer creation. We can
