diff --git ui/views/controls/menu/menu_host.cc ui/views/controls/menu/menu_host.cc
index 0a8cb87554c2f..bcf118fd2cb7a 100644
--- ui/views/controls/menu/menu_host.cc
+++ ui/views/controls/menu/menu_host.cc
@@ -157,7 +157,11 @@ void MenuHost::InitMenuHost(const InitParams& init_params) {
   // get keyboard focus.
   if (init_params.parent == nullptr &&
       init_params.parent_widget == gfx::kNullAcceleratedWidget)
+#if defined(OS_LINUX)
+    params.activatable = Widget::InitParams::Activatable::kNo;
+#else
     params.activatable = Widget::InitParams::Activatable::kYes;
+#endif

 #if BUILDFLAG(IS_WIN)
   // On Windows use the software compositor to ensure that we don't block
